sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/InternalRow.scala

```diff
@@ -119,4 +119,28 @@ object InternalRow {
     case v: MapData => v.copy()
     case _ => value
   }
+
+  /**
+   * Returns an accessor for an `InternalRow` with given data type. The returned accessor
+   * actually takes a `SpecializedGetters` input because it can be generalized to other classes
+   * that implements `SpecializedGetters` (e.g., `ArrayData`) too.
+   */
+  def getAccessor(dataType: DataType): (SpecializedGetters, Int) => Any = dataType match {
```

Comment:
Perhaps we should move this to the companion object of `SpecializedGetters`?

Reply:
Ok.

Reply:
As `SpecializedGetters` is a java interface, seems we can't create a companion object for it? Or I miss something?

Reply:
Hehe - good point. Let's leave it here for now.